src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -297,11 +298,13 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     }
     if (_.isInteger(this.autoReload) && this.autoReload > 0) {
       this.ngZone.runOutsideAngular(() => {
-        this.reloadSubscriber = observableTimer(0, this.autoReload).subscribe(() => {
-          this.ngZone.run(() => {
-            return this.reloadData();
+        this.reloadSubscriber = observableTimer(0, this.autoReload)
+          .pipe(whenPageVisible())
+          .subscribe(() => {
+            this.ngZone.run(() => {
+              return this.reloadData();
+            });
```

Comment:
What about refactoring this using the `timer()` service below as most other polling services do (since it's basically doing the same: ngZone, timer, ...)?

Reply:
Code Updated!